content/python/concepts/memoization/memoization.md

```diff
@@ -0,0 +1,77 @@
+---
+Title: 'Memoization'
+Description: 'Memoization can be used to improve performance in Python code.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Conceptual'
+  - 'Dynamic Programming'
+CatalogContent:
+  - 'learn-data-structures-and-algorithms-with-python'
+  - 'paths/computer-science'
+---
+
+In Python, [memoization](https://www.codecademy.com/resources/docs/general/memoization) can be used to improve a program's performance.
+
+This is ideal whenever the same calculations (with the same inputs and outputs) are known to take place. Memoization helps by saving, or caching, these results in computer memory for later use. It can be applied to many contexts, including:
+
+- Making calls to recursive or pure function where some calculations use the same inputs and return the same outputs.
+- Fetching from the server with the same base API endpoints.
+
+## Syntax
+
+```pseudo
+memo = {}
+
+def memo_function(args):
+  if(args in memo):
+    return memo[args]
+  else:
+    # Do this and update memo.
+
+```
+
+In Python, one way memoized results are represented is with [dictionaries](https://www.codecademy.com/resources/docs/python/dictionaries). In a given function, if a previously calculated result exists in the `memo` dictionary, it is returned. Otherwise, normal calculations take place and the `memo` is updated.
```

Comment:
```suggestion
In Python, one way memoized results are represented is with [dictionaries](https://www.codecademy.com/resources/docs/python/dictionaries). In a given function, if a previously calculated result exists in the `memo` dictionary, it is returned. Otherwise, normal calculations take place and the `memo` dictionary is updated.
```